src/auth/views/Login.tsx

```diff
@@ -1,20 +1,89 @@
+import { APP_MOUNT_URI } from "@saleor/config";
 import useNavigator from "@saleor/hooks/useNavigator";
 import useUser from "@saleor/hooks/useUser";
-import React from "react";
+import React, { useEffect, useState } from "react";
+import { useQuery } from "react-apollo";
+import urlJoin from "url-join";
+import useRouter from "use-react-router";
 
-import LoginPage, { FormData } from "../components/LoginPage";
-import { passwordResetUrl } from "../urls";
+import LoginPage from "../components/LoginPage";
+import { LoginFormData } from "../components/LoginPage/form";
+import { availableExternalAuthentications } from "../queries";
+import { AvailableExternalAuthentications } from "../types/AvailableExternalAuthentications";
+import {
+  loginCallbackPath,
+  LoginUrlQueryParams,
+  passwordResetUrl
+} from "../urls";
 
-const LoginView: React.FC = () => {
+interface LoginViewProps {
+  params: LoginUrlQueryParams;
+}
+
+const LoginView: React.FC<LoginViewProps> = ({ params }) => {
   const navigate = useNavigator();
-  const { login, user, tokenAuthLoading } = useUser();
+  const { location } = useRouter();
+  const {
+    login,
+    requestLoginByExternalPlugin,
+    loginByExternalPlugin,
+    tokenAuthLoading
+  } = useUser();
+  const [isError, setIsError] = useState(false);
+  const [isExternalError, setIsExternalError] = useState(false);
+  const { data: externalAuthentications } = useQuery<
+    AvailableExternalAuthentications
+  >(availableExternalAuthentications);
+
+  const handleSubmit = async (data: LoginFormData) => {
+    const result = await login(data.email, data.password);
+    const errors = result?.errors || [];
+
+    setIsExternalError(false);
+    setIsError(!result || errors?.length > 0);
+    return errors;
+  };
+
+  const handleRequestExternalAuthentication = (pluginId: string) =>
+    requestLoginByExternalPlugin(pluginId, {
+      redirectUri: urlJoin(
+        window.location.origin,
+        APP_MOUNT_URI === "/" ? "" : APP_MOUNT_URI,
```

Comment:
Can we put "/" to const like DEFAULT_URI ?

Reply:
sure 👍